ggshield/cmd/secret/ignore.py

```diff
@@ -7,41 +7,74 @@
 from ggshield.core.cache import Cache
 from ggshield.core.config import Config
 from ggshield.core.text_utils import pluralize
+from ggshield.core.types import IgnoredMatch
 
 
 @click.command()
+@click.argument(
+    "secret_sha",
+    nargs=1,
+    type=str,
+    required=False,
+    metavar="SECRET_SHA",
+)
 @click.option(
     "--last-found",
     is_flag=True,
     help="Ignore secrets found in the last `ggshield secret scan` run.",
 )
+@click.option(
+    "--name",
+    type=str,
+    help="Name of the secret to ignore.",
+    metavar="NAME",
+)
 @add_common_options()
 @click.pass_context
 def ignore_cmd(
     ctx: click.Context,
+    secret_sha: str,
+    name: str,
     last_found: bool,
     **kwargs: Any,
 ) -> None:
     """
     Ignore some secrets.
 
-    The `secret ignore` command instructs ggshield to ignore secrets it finds during a scan.
+    The `secret ignore` command instructs ggshield to ignore secrets.
 
-    This command needs to be used with an option to determine which secrets it should ignore.
-    For now, it only handles the `--last-found` option, which ignores all secrets found during the last scan.
+    Option `--name` allows to specify the name of the secret to ignore.
+
+    Option `--last-found` ignores all secrets found during the last scan.
 
     The command adds the ignored secrets to the `secrets.ignored-matches` section of your local
     configuration file. If no local configuration file is found, a `.gitguardian.yaml` file is created.
     """
+
+    ctx_obj = ContextObj.get(ctx)
+    config = ctx_obj.config
+    path = config.config_path
+
     if last_found:
-        ctx_obj = ContextObj.get(ctx)
-        config = ctx_obj.config
+        if secret_sha or name:
+            raise click.UsageError(
+                "Option `--last-found` cannot be used with `SECRET_SHA` or `--name`."
+            )
         nb = ignore_last_found(config, ctx_obj.cache)
-        path = config.config_path
-        secrets_word = pluralize("secret", nb)
-        click.echo(
-            f"Added {nb} {secrets_word} to the `secret.ignored-matches` section of {path}."
-        )
+    else:
+        if not name:
+            raise click.UsageError(
+                "Option `--name` is required when ignoring a secret."
+            )
+        ignored_match = IgnoredMatch(name=name if name else "", match=secret_sha)
+        config.add_ignored_match(ignored_match)
+        nb = 1
+
+    config.save()
+    secrets_word = pluralize("secret", nb)
+    click.echo(
+        f"Added {nb} {secrets_word} to the secret.ignored-matches section of {path}."
```

Comment:
Sorry, I am going to do a "while you are at it"...

Can you replace `ignored-matches` with `ignored_matches`?